app/models/user.rb

```diff
@@ -5,9 +5,15 @@ class User < ActiveRecord::Base
 
   validates :name, :email, presence: true, uniqueness: true
   validates :email, confirmation: true
-  validates :password, length: { in: 6..64 }
+  validates :password, length: { in: 6..64, if: :validate_password? }
```

Comment:
A suggestion to change this line to:
` validates :password, length: { in: 6..64 }, allow_nil: true`

Reply:
Fantastic solution, thank you, done in sha : 27867eb